element-connector/src/main/java/org/eclipse/californium/elements/EndpointContext.java

```diff
@@ -14,31 +14,87 @@
  *    Bosch Software Innovations GmbH - add support for correlation context to provide
  *                                      additional information to application layer for
  *                                      matching messages (fix GitHub issue #1)
+ *    Achim Kraus (Bosch Software Innovations GmbH) - extend and rename 
+ *                                      CorrelationContext into EndpointContext.
  ******************************************************************************/
 package org.eclipse.californium.elements;
 
+import java.net.InetSocketAddress;
+import java.security.Principal;
 import java.util.Map;
 import java.util.Set;
 
 /**
  * A container for storing transport specific information about the context in
  * which a message has been sent or received.
+ * 
+ * Contains several endpoint identity candidates:
+ * 
+ * <table summary="Endpoint Identity Candidates">
+ * <tr>
+ * <th>Candidate</th>
+ * <th>Description</th>
+ * <th>Long-Term-Stable</th>
+ * </tr>
+ * <tr>
+ * <td>InetSocketAddress</td>
+ * <td>identity for plain coap according RFC7252</td>
+ * <td>With IPv4 and NATs too frequently not long-term-stable</td>
+ * </tr>
+ * <tr>
+ * <td>Principal</td>
+ * <td>identity for secure endpoint, requires unique credentials per client</td>
+ * <td>As stable as the credentials</td>
+ * </tr>
+ * <tr>
+ * <td>DTLS Session, cipher suite, epoch</td>
+ * <td>identity for secure coaps according RFC7252</td>
+ * <td>With new or resumed DTLS session not long-term-stable</td>
+ * </tr>
+ * </table>
  */
 public interface EndpointContext {
 
 	/**
 	 * Gets a value from this context.
 	 * 
 	 * @param key the key to retrieve the value for.
-	 * @return the value or <code>null</code> if this context does not contain a
+	 * @return the value or {@code null} if this context does not contain a
 	 *         value for the given key.
 	 */
 	String get(String key);
 
 	/**
-	 * Gets a Set of a Map.Entry which contains the key-value pair of the EndpointContext.
+	 * Gets a Set of a Map.Entry which contains the key-value pair of the
+	 * CorrelationContext.
+	 * 
+	 * The Set is intended to be "unmodifiable".
 	 *
 	 * @return A set of a map entry containing the key value pair.
 	 */
 	Set<Map.Entry<String, String>> entrySet();
+
+	/**
+	 * Check, if the correlation information contained, will inhibit a new
+	 * connection.
+	 * 
+	 * @return {@code true}, if no new connection could match the correlation
+	 *         context provided in this instance. {@code false}, if a new
+	 *         connection may match this correlation context.
+	 */
+	boolean inhibitNewConnection();
```

Comment:
I do not understand the purpose of this method.

Reply:
Sending a message using a connector sometimes "triggers" a connection of handshake. If the EndpointContext provides data in the endpoint context, which excludes new connections or handshakes to bb used succesful for sending, you can drop the message early.

E.g. if "strict" DTLS session based matching is wanted and a message for a defined dtls session is send, it would make sense to start a new handshake, because, it would not get the provided session when it gets a new session. Or if a response is sent for coap+tcp but the connection is dropped, it doesn't make sense to establish a new connection.

Reply:
ok